.github/workflows/env-heaan.yml

```diff
@@ -0,0 +1,231 @@
+        id: sha
+        run: echo "::set-output name=sha::$(git -C src/z3 rev-parse HEAD)"
+
+      - name: Fetch Z3 cache
+        id: cache
+        uses: actions/cache@v2
+        with:
+          path: /tmp/image/opt/z3
+          key: cache-z3-${{ secrets.CACHE_VERSION }}-${{ steps.sha.outputs.sha }}
+
+      # if cache miss
+      - name: Fetch dependency cache
+        if: steps.cache.outputs.cache-hit != 'true'
+        uses: actions/cache@v2
+        with:
+          path: /tmp/image/.dep
+          key: cache-dep-${{ secrets.CACHE_VERSION }}
+
+      - name: Unpack dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          sudo apt install /tmp/image/.dep/*.deb
+
+      - name: Rebuild Z3
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          mkdir -p /tmp/image/opt/z3
+          cd src/z3
+          cmake -B build -G Ninja \
+            -DCMAKE_INSTALL_PREFIX=/tmp/image/opt/z3 -DCMAKE_BUILD_TYPE=MinSizeRel
+          cmake --build build --target install
+      # endif cache miss
+
+  configure-cvc5:
+    if: github.repository == 'aqjune/mlir-tv'
+    needs: configure-dependencies
+    runs-on: ubuntu-20.04
+    outputs:
+      sha: ${{ steps.sha.outputs.sha }}
+    steps:
+      - name: Checkout CVC5
+        uses: actions/checkout@v2
+        with:
+          repository: cvc5/cvc5
+          ref: cvc5-0.0.3
+          path: src/cvc5
+
+      - name: Get CVC5 SHA
+        id: sha
+        run: echo "::set-output name=sha::$(git -C src/cvc5 rev-parse HEAD)"
+
+      - name: Fetch CVC5 cache
+        id: cache
+        uses: actions/cache@v2
+        with:
+          path: /tmp/image/opt/cvc5
+          key: cache-cvc5-${{ secrets.CACHE_VERSION }}-${{ steps.sha.outputs.sha }}
+
+      # if cache miss
+      - name: Fetch dependency cache
+        if: steps.cache.outputs.cache-hit != 'true'
+        uses: actions/cache@v2
+        with:
+          path: /tmp/image/.dep
+          key: cache-dep-${{ secrets.CACHE_VERSION }}
+
+      - name: Unpack dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          sudo apt install /tmp/image/.dep/*.deb          
+      
+      - name: Rebuild CVC5
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          sudo python3 -m pip install toml
+          mkdir -p /tmp/image/opt/cvc5
+          cd src/cvc5
+          ./configure.sh --prefix=/tmp/image/opt/cvc5 --ninja --no-poly --auto-download
+          ninja -C build
+          ninja -C build install
+      # endif cache miss
+
+  configure-llvm:
+    if: github.repository == 'aqjune/mlir-tv'
+    needs: configure-dependencies
+    runs-on: ubuntu-20.04
+    outputs:
+      sha: ${{ steps.sha.outputs.sha }}
+    steps:
+      - name: Checkout LLVM
+        uses: actions/checkout@v2
+        with:
+          repository: llvm/llvm-project
+          path: src/llvm-project
+          ref: 27bb799095c1dfd2e83fe0eabf6399fa83082444
```

Comment:
Is it possible to set the LLVM's branch name to `heaan-mlir` instead of designating a commit id?

Reply:
Yes, we can use branch name instead of commit hash. Willfix.

Reply:
Fixed